std/sys/Http.hx

```diff
@@ -58,12 +58,20 @@ class Http extends haxe.http.HttpBase {
 			onError(e);
 		}
 		customRequest(post,output);
-		if(!err)
+		if(!err){
 		#if neko
 			onData(responseData = neko.Lib.stringReference(output.getBytes()));
 		#else
-			onData(responseData = output.getBytes().toString());
+			var respType = responseHeaders.get("Accept-Ranges");
```

Comment:
I'm not sure checking Accept-Ranges is the right approach to determine if the response is binary – this just implies the resource can be accessed with range requests (which many binary resources will not set)

For the most predictable behavior we need to mirror whatever the industry standard is as closely as possible. Checking the "Content-Type" header would be the normal approach (details here: https://tools.ietf.org/html/rfc7231#section-3.1.1.5), however if this is not set we need to choose a sensible guess. The specification says
> If a Content-Type header field is not present, the recipient
   MAY either assume a media type of "application/octet-stream"
   ([RFC2046], Section 4.5.1) or examine the data to determine its type.

But I'd want to review what popular implementations are actually doing (like web browsers) so we're conforming to everyone's expectations.